src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/GameManager.java

```diff
@@ -980,6 +980,48 @@ public List<GameboardItem> getNextQuestionsForFilter(final GameFilter gameFilter
     return selectionOfGameboardQuestions;
   }
 
+  /**
+   * This generates a set of question in random.
+   *
+   * @param gameFilter - to enable search
+   * @param limit      - to provide a limit of questions
+   * @return a list of questions
+   * @throws ContentManagerException - if there is a problem accessing the content repository.
+   */
+  public List<IsaacQuestionPageDTO> generateRandomQuestions(final GameFilter gameFilter, final int limit)
+      throws ContentManagerException {
+    List<GitContentManager.BooleanSearchClause> fieldsToMap = Lists.newArrayList();
+
+    fieldsToMap.add(new GitContentManager.BooleanSearchClause(TYPE_FIELDNAME, BooleanOperator.AND,
+        Collections.singletonList(QUESTION_TYPE)));
+
+    // exclude questions marked deprecated
+    fieldsToMap.add(new GitContentManager.BooleanSearchClause(DEPRECATED_FIELDNAME, BooleanOperator.NOT,
```

Comment:
Have you **manually** tested the following scenarios?

1. The object stored in Elasticsearch does not have a `deprecated` field.
2. The object stored in Elasticsearch has a `deprecated` field with a `true` value.
3. The object stored in Elasticsearch has a `deprecated` field with a `false` value.

If so, what happens in every scenario? I would expect the object to be returned in scenarios 1 and 3, but not in 2.

Reply:
Manual testing was done for deprecated (which is always either true or no value) and supersededBy (which is a string or no value).

1. If deprecated is true, the question is not returned 
2. If supersededBy has a value, the question is not returned
3. If neither of those are true (i.e. there is no supersededBy and no deprecated), the question is returned. 

Where no questions matched the requirements, an empty array is returned which allows the frontend to fallback to an alternative component.